src/Features/CSharp/Portable/InlineHints/CSharpInlineTypeHintsService.cs

```diff
@@ -42,7 +42,14 @@ public CSharpInlineTypeHintsService()
                     if (IsValidType(type))
                         return (type, GetSpan(displayAllOverride, forImplicitVariableTypes, variableDeclaration.Type, variableDeclaration.Variables[0].Identifier));
                 }
-                else if (node is SingleVariableDesignationSyntax { Parent: not DeclarationPatternSyntax } variableDesignation)
+                if (node is DeclarationExpressionSyntax declarationExpression &&
+                    declarationExpression.Type.IsVar)
```

Comment:
Have the option of using a pattern here too:

```suggestion
                if (node is DeclarationExpressionSyntax { Type: { IsVar: true } })
```

Reply:
yup.  will do :)